WalletWasabi/Crypto/ZeroKnowledge/ZkProver.cs

```diff
@@ -0,0 +1,24 @@
+using NBitcoin.Secp256k1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using WalletWasabi.WabiSabi.Crypto;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public static class ZkProver
+	{
+		public static ZkProof CreateProof(Scalar exponent)
+		{
+			var publicPoint = (EC.G * exponent).ToGroupElement();
+			var randomScalar = SecureRandom.GetScalarNonZero();
+			var randomPoint = (EC.G * randomScalar).ToGroupElement();
+			var challenge = ZkChallenge.Build(publicPoint, randomPoint);
+
+			var response = randomScalar + exponent * challenge;
+
+			return new ZkProof(publicPoint, randomPoint, response);
```

Comment:
This looks wrong to me. I have read lots and lots of papers about this topic and a proof is always `(R, s)` or in this case `(publicPoint, response)`. You cannot send the `randomPoint` because it is secret in the same way `exponent` is secret.

Reply:
Forget it. My mistake, I misunderstood it because the notation was confusing to me. This is okay.